<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Variant Tracker • outbreakinfo</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Variant Tracker">
<meta property="og:description" content="outbreakinfo">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-VK9KSJ8ENE"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-VK9KSJ8ENE');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">outbreakinfo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/outbreakinfo.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/epivignette.html">Cases &amp; Deaths data</a>
    </li>
    <li>
      <a href="../articles/locationtracker.html">Location Tracker</a>
    </li>
    <li>
      <a href="../articles/researchlibrary.html">Research Library</a>
    </li>
    <li>
      <a href="../articles/varianttracker.html">Variant Tracker</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/outbreak-info/R-outbreak-info/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="varianttracker_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Variant Tracker</h1>
            
            <h4 data-toc-skip class="date">24 June 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/outbreak-info/R-outbreak-info/blob/HEAD/vignettes/varianttracker.Rmd" class="external-link"><code>vignettes/varianttracker.Rmd</code></a></small>
      <div class="hidden name"><code>varianttracker.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="accessing-the-data-in-outbreak-infos-variant-tracker">Accessing the data in outbreak.info’s Variant Tracker<a class="anchor" aria-label="anchor" href="#accessing-the-data-in-outbreak-infos-variant-tracker"></a>
</h2>
<p>outbreak.info’s Variant Tracker allows you to view how combinations of lineages and mutations are changing over time. These reports allow you to answer questions like:</p>
<ul>
<li><a href="#what-mutations-are-characteristic-of-the-variant-">What mutations are present within that lineage?</a></li>
<li><a href="#how-prevalent-is-a-variant-overall-">How prevalent is a variant overall?</a></li>
<li><a href="#how-is-the-prevalence-of-a-variant-changing-">How is the prevalence of the variant changing?</a></li>
<li><a href="#where-is-the-variant-most-prevalent-">Where is the variant most prevalent?</a></li>
<li><a href="#what-do-we-know-about-the-variant-in-the-literature-">What is known about the variant in the research literature?</a></li>
</ul>
<p>Here, we’ll demonstrate how to access the data in the <a href="https://outbreak.info/situation-reports?pango=P.1&amp;loc=BRA&amp;loc=USA&amp;selected=Worldwide&amp;overlay=false" class="external-link">P.1 Lineage Report</a>. The P.1 / Gamma lineage was labeled a <a href="https://outbreak.info/situation-reports" class="external-link">Variant of Concern</a> by the WHO in January 2021 because it showed evidence of increased transmissibility, virulence, and/or decreased diagnostic, therapeutic, or vaccine efficacy.</p>
<div class="section level3">
<h3 id="before-we-start">Before we start…<a class="anchor" aria-label="anchor" href="#before-we-start"></a>
</h3>
<p>Import the packages we’ll use and provide our GISAID credentials to access the data. If you don’t have a GISAID account, you can <a href="https://www.gisaid.org/registration/register/" class="external-link">register</a> for one on their website. It may take a day or two for the account to become active.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Package imports</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://purrr.tidyverse.org" class="external-link">purrr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/" class="external-link">knitr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://outbreak-info.github.io/R-outbreak-info/" class="external-link">outbreakinfo</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: replacing previous import 'jsonlite::flatten' by 'purrr::flatten' when</span>
<span class="co">#&gt; loading 'outbreakinfo'</span>
<span class="co">#&gt; Warning: replacing previous import 'ggplot2::%+%' by 'crayon::%+%' when loading</span>
<span class="co">#&gt; 'outbreakinfo'</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Authenticate yourself using your GISAID credentials.</span>
<span class="fu"><a href="../reference/authenticateUser.html">authenticateUser</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="what-mutations-are-characteristic-of-the-variant">What mutations are characteristic of the variant?<a class="anchor" aria-label="anchor" href="#what-mutations-are-characteristic-of-the-variant"></a>
</h3>
<p>A very basic question: what is P.1? What mutations consistently appear in most sequences within the lineage?</p>
<div class="section level4">
<h4 id="grabbing-the-characteristic-mutations">Grabbing the characteristic mutations<a class="anchor" aria-label="anchor" href="#grabbing-the-characteristic-mutations"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="../reference/getMutationsByLineage.html">getMutationsByLineage</a></span><span class="op">(</span>pangolin_lineage<span class="op">=</span><span class="st">"P.1"</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span>
<span class="fu"><a href="../reference/plotMutationHeatmap.html">plotMutationHeatmap</a></span><span class="op">(</span><span class="va">p1</span>, title<span class="op">=</span><span class="st">"Characteristic mutations of P.1 occurring in at least 75% of sequences"</span><span class="op">)</span></code></pre></div>
<p><img src="varianttracker_files/figure-html/characteristic_mutations-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="comparing-to-other-lineages">Comparing to other lineages<a class="anchor" aria-label="anchor" href="#comparing-to-other-lineages"></a>
</h4>
<p>We can also compare how the mutations of P.1 compare to its sublineages, to look how any mutations may have been aquired or disappeared. We can also lower (or raise) the threshold for which mutations are included</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># First: find which sublineages are associated with P.1:</span>
<span class="va">gamma_lineages</span> <span class="op">=</span> <span class="fu"><a href="../reference/lookupSublineages.html">lookupSublineages</a></span><span class="op">(</span><span class="st">"Gamma"</span>, returnQueryString <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># Collect the data</span>
<span class="va">gamma_mutations</span> <span class="op">=</span> <span class="fu"><a href="../reference/getMutationsByLineage.html">getMutationsByLineage</a></span><span class="op">(</span>pangolin_lineage<span class="op">=</span><span class="va">gamma_lineages</span>, frequency<span class="op">=</span><span class="fl">0.5</span>, logInfo <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># Plot!</span>
<span class="fu"><a href="../reference/plotMutationHeatmap.html">plotMutationHeatmap</a></span><span class="op">(</span><span class="va">gamma_mutations</span>, title <span class="op">=</span> <span class="st">"S-gene mutations in &gt; 50% of Gamma sequences"</span><span class="op">)</span></code></pre></div>
<p><img src="varianttracker_files/figure-html/all_gamma_mutations-1.png" width="700"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># You can also change the gene you select by changing</span>
<span class="co"># plotMutationHeatmap(gamma_mutations, title = "ORF1a-gene mutations in Gamma lineages", gene2Plot="ORF1a")</span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="how-prevalent-is-a-variant-overall">How prevalent is a variant overall?<a class="anchor" aria-label="anchor" href="#how-prevalent-is-a-variant-overall"></a>
</h3>
<p>To start with, let’s get some basic information about the overall, cumulative prevalence of P.1 in certain locations. We’ll look at the worldwide totals, in Brazil, in the United States, and in Illinois.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">locations</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Brazil"</span>, <span class="st">"United States"</span>, <span class="st">"Illinois"</span><span class="op">)</span>

<span class="va">worldwide_prevalence</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPrevalence.html">getPrevalence</a></span><span class="op">(</span>pangolin_lineage <span class="op">=</span> <span class="st">"P.1"</span>, cumulative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span>
<span class="va">cumulative_prevalences</span> <span class="op">=</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_df</a></span><span class="op">(</span><span class="va">locations</span>, <span class="kw">function</span><span class="op">(</span><span class="va">loc</span><span class="op">)</span> <span class="fu"><a href="../reference/getPrevalence.html">getPrevalence</a></span><span class="op">(</span>pangolin_lineage <span class="op">=</span> <span class="st">"P.1"</span>, location <span class="op">=</span> <span class="va">loc</span>, cumulative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">value.global_prevalence</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data... </span>
<span class="co">#&gt; Retrieving data... </span>
<span class="co">#&gt; Retrieving data...</span>

<span class="va">cumulative_prevalences</span> <span class="op">=</span> <span class="va">worldwide_prevalence</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">cumulative_prevalences</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">cumulative_prevalences</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">location</span>, <span class="va">value.global_prevalence</span>, <span class="va">value.lineage_count</span>, <span class="va">value.first_detected</span>, <span class="va">value.last_detected</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="14%">
<col width="24%">
<col width="20%">
<col width="21%">
<col width="20%">
</colgroup>
<thead><tr class="header">
<th align="left">location</th>
<th align="right">value.global_prevalence</th>
<th align="right">value.lineage_count</th>
<th align="left">value.first_detected</th>
<th align="left">value.last_detected</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Worldwide</td>
<td align="right">0.0062128</td>
<td align="right">68722</td>
<td align="left">2020-04-07</td>
<td align="left">2021-12-30</td>
</tr>
<tr class="even">
<td align="left">Brazil</td>
<td align="right">0.2514963</td>
<td align="right">37355</td>
<td align="left">2020-05-06</td>
<td align="left">2021-12-30</td>
</tr>
<tr class="odd">
<td align="left">Illinois</td>
<td align="right">0.0446179</td>
<td align="right">3556</td>
<td align="left">2020-04-07</td>
<td align="left">2021-08-20</td>
</tr>
<tr class="even">
<td align="left">United States</td>
<td align="right">0.0057024</td>
<td align="right">19761</td>
<td align="left">2020-04-07</td>
<td align="left">2021-12-09</td>
</tr>
</tbody>
</table>
<hr>
</div>
<div class="section level3">
<h3 id="how-is-the-prevalence-of-a-variant-changing">How is the prevalence of a variant changing?<a class="anchor" aria-label="anchor" href="#how-is-the-prevalence-of-a-variant-changing"></a>
</h3>
<p>We can access how the prevalence of a lineage, a group of lineages, a mutation, a group of mutations, or a lineage with additional mutations is changing, customizable to any of the over 2,000 countries, states/provinces, or U.S. counties.</p>
<div class="section level4">
<h4 id="lineage-queries">Lineage queries<a class="anchor" aria-label="anchor" href="#lineage-queries"></a>
</h4>
<p>Let’s start with something simple: just the prevalence of P.1 across Brazil, where we see it becomes dominant in spring 2021 but then falls to near 0 prevalence.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1_brazil</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPrevalence.html">getPrevalence</a></span><span class="op">(</span>pangolin_lineage <span class="op">=</span> <span class="st">"P.1"</span>, location <span class="op">=</span> <span class="st">"Brazil"</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span>
<span class="fu"><a href="../reference/plotPrevalenceOverTime.html">plotPrevalenceOverTime</a></span><span class="op">(</span><span class="va">p1_brazil</span>, title <span class="op">=</span> <span class="st">"Daily P.1 prevalence in Brazil"</span><span class="op">)</span></code></pre></div>
<p><img src="varianttracker_files/figure-html/p1-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="comparing-lineage-queries-multiple-lineage-queries-">Comparing lineage queries &amp; multiple lineage queries.<a class="anchor" aria-label="anchor" href="#comparing-lineage-queries-multiple-lineage-queries-"></a>
</h4>
<p>How does that compare to the appearance of Delta lineages in Brazil? Or to the proliferation of P.1 in the United States? Using the same functions with slightly different parameters, we can grab the data for the Delta lineages in Brazil and P.1 in the U.S. and see how they compare.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Multiple lineage queries should be separated by the parameter ` OR `. The `lookupSublinage` function is a convenient wrapper to look up what sublineages are associated with Delta (e.g. B.1.617.2 and all the AY sublineages) and collapse it into a string to be used in `getPrevalence`.</span>
<span class="va">delta_lineages_string</span> <span class="op">=</span> <span class="fu"><a href="../reference/lookupSublineages.html">lookupSublineages</a></span><span class="op">(</span><span class="st">"Delta"</span>, returnQueryString <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># the name of this lineage gets REALLY long, so we can change the name with a named dictionary:</span>
<span class="va">delta_label</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Delta"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">delta_label</span><span class="op">)</span> <span class="op">=</span> <span class="va">delta_lineages_string</span>

<span class="co"># Get Delta prevalence, combine into a single dataset, and plot</span>
<span class="va">delta_brazil</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPrevalence.html">getPrevalence</a></span><span class="op">(</span>pangolin_lineage <span class="op">=</span> <span class="va">delta_lineages_string</span>, location <span class="op">=</span> <span class="st">"Brazil"</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span>
<span class="va">p1_delta_brazil</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">p1_brazil</span>, <span class="va">delta_brazil</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotPrevalenceOverTime.html">plotPrevalenceOverTime</a></span><span class="op">(</span><span class="va">p1_delta_brazil</span>, title <span class="op">=</span> <span class="st">"Daily P.1 and Delta prevalence in Brazil"</span>, labelDictionary <span class="op">=</span> <span class="va">delta_label</span><span class="op">)</span></code></pre></div>
<p><img src="varianttracker_files/figure-html/p1_comparison-1.png" width="700"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">

<span class="co"># Comparing P.1 in the U.S. and Brazil. We also need to change what variable colors the traces.</span>
<span class="va">p1_usa</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPrevalence.html">getPrevalence</a></span><span class="op">(</span>pangolin_lineage <span class="op">=</span> <span class="st">"P.1"</span>, location <span class="op">=</span> <span class="st">"United States"</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span>
<span class="va">p1</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">p1_brazil</span>, <span class="va">p1_usa</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotPrevalenceOverTime.html">plotPrevalenceOverTime</a></span><span class="op">(</span><span class="va">p1</span>, title <span class="op">=</span> <span class="st">"Daily P.1 prevalence"</span>, colorVar <span class="op">=</span> <span class="st">"location"</span><span class="op">)</span></code></pre></div>
<p><img src="varianttracker_files/figure-html/p1_comparison-2.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="mutation-queries">Mutation queries<a class="anchor" aria-label="anchor" href="#mutation-queries"></a>
</h4>
<p>From the <a href="#grabbing-the-characteristic-mutations">Characteristic mutations</a> plots, two mutations of interest that pop up in the P.1 and Gamma sequences are N501Y and E484K in the Spike protein. Rather than look at the prevalence of a Pango lineage, you can also look at how the prevalence of these two mutations is changing. Be sure to add the gene in front of the mutation, like <code>S:N501Y</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># With no location specified, the results for all samples (Worldwide) are returned.</span>
<span class="va">n501y_e484k</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPrevalence.html">getPrevalence</a></span><span class="op">(</span>mutations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S:E484K"</span>, <span class="st">"S:N501Y"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span>
<span class="fu"><a href="../reference/plotPrevalenceOverTime.html">plotPrevalenceOverTime</a></span><span class="op">(</span><span class="va">n501y_e484k</span>, title <span class="op">=</span> <span class="st">"Daily S:E484K &amp; S:N501Y prevalence"</span><span class="op">)</span></code></pre></div>
<p><img src="varianttracker_files/figure-html/mutation_queries-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># You can also see how often those two combinations of mutations appear in lineages:</span>
<span class="va">lineages_n501y_e484k</span> <span class="op">=</span> <span class="fu"><a href="../reference/getMutationAcrossLineage.html">getMutationAcrossLineage</a></span><span class="op">(</span>mutations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S:E484K"</span>, <span class="st">"S:N501Y"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span>

<span class="co"># We'll sort those lineages and take the top 5, of lineages with at least 1,000 sequences:</span>
<span class="va">lineages_n501y_e484k</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">pangolin_lineage</span>, <span class="va">proportion</span>, <span class="va">lineage_count</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">lineage_count</span> <span class="op">&gt;=</span> <span class="fl">1000</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">proportion</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">pangolin_lineage</th>
<th align="right">proportion</th>
<th align="right">lineage_count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ba.2.24</td>
<td align="right">0.9994292</td>
<td align="right">3504</td>
</tr>
<tr class="even">
<td align="left">ba.2.29</td>
<td align="right">0.9992724</td>
<td align="right">4123</td>
</tr>
<tr class="odd">
<td align="left">ba.2.3.1</td>
<td align="right">0.9972887</td>
<td align="right">6270</td>
</tr>
<tr class="even">
<td align="left">ba.2.2</td>
<td align="right">0.9964893</td>
<td align="right">3703</td>
</tr>
<tr class="odd">
<td align="left">ba.1.1.9</td>
<td align="right">0.9953364</td>
<td align="right">1501</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># ... and you can combine mutations with a Pango lineage to look at how those mutations are prevalent in the P.1 backbone.</span>
<span class="va">P1_n501y_e484k</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPrevalence.html">getPrevalence</a></span><span class="op">(</span>mutations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S:E484K"</span>, <span class="st">"S:N501Y"</span><span class="op">)</span>, pangolin_lineage <span class="op">=</span> <span class="st">"P.1"</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span></code></pre></div>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="where-is-the-variant-most-prevalent">Where is the variant most prevalent?<a class="anchor" aria-label="anchor" href="#where-is-the-variant-most-prevalent"></a>
</h3>
<p>After seeing how P.1 prevalence changes in specific locations, a natural related question is where is the prevalence of P.1 the greatest? We can access that data for the entire world, to look at how the prevalence varies by country, or we can look at how it varies within a country like Brazil.</p>
<div class="section level4">
<h4 id="getting-p-1-cumulative-prevalence-by-country">Getting P.1 cumulative prevalence by country<a class="anchor" aria-label="anchor" href="#getting-p-1-cumulative-prevalence-by-country"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1_by_country</span> <span class="op">=</span> <span class="fu"><a href="../reference/getCumulativeBySubadmin.html">getCumulativeBySubadmin</a></span><span class="op">(</span>pangolin_lineage <span class="op">=</span> <span class="st">"P.1"</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span></code></pre></div>
<pre><code><span class="co">#&gt; Linking to GEOS 3.8.0, GDAL 3.0.4, PROJ 6.3.1; sf_use_s2() is TRUE</span></code></pre>
</div>
<div class="section level4">
<h4 id="making-a-choropleth">Making a choropleth<a class="anchor" aria-label="anchor" href="#making-a-choropleth"></a>
</h4>
<p>The location names in the genomics data are standardized to the location ids contained in the <a href="https://gadm.org/download_world.html" class="external-link">GADM data</a>. After downloading the shapefile from GADM, it’s easy to make a choropleth using the <a href="https://r-spatial.github.io/sf/" class="external-link">sf</a> package. Note: since GADM doesn’t allow their data to be redistributed, you’ll need to download the data yourself. But it’s free to use! You might also want to run the data through <a href="https://mapshaper.org/" class="external-link">mapshaper</a> to simplify the geometries, since the data is quite large. Within the choropleth, it’s easy to see that P.1 has been more prevalent across South America.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-spatial.github.io/sf/" class="external-link">sf</a></span><span class="op">)</span>
<span class="co"># You'll need to download and import the GADM Administrative level 1 data, with something like:</span>
<span class="co"># gadm = read_sf("gadm36_0.shp")</span>

<span class="co"># `gadm` is an sf object containing the GADM basemap. Geojoin the data by the ISO3 ID</span>
<span class="va">gadm_p1</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">gadm</span>, <span class="va">p1_by_country</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GID_0"</span> <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Use the basic plotting function</span>
<span class="fu"><a href="../reference/plotChoropleth.html">plotChoropleth</a></span><span class="op">(</span><span class="va">gadm_p1</span>, title<span class="op">=</span><span class="st">"Cumulative P.1 prevalence"</span><span class="op">)</span></code></pre></div>
<p><img src="varianttracker_files/figure-html/world_choropleth-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="adjusting-the-time-window">Adjusting the time window<a class="anchor" aria-label="anchor" href="#adjusting-the-time-window"></a>
</h4>
<p>The choropleth above shows the cumulative prevalence of P.1 since it was first detected. However, more recently, its prevalence has dropped. We can adjust this time window by modifying the <code>ndays</code> parameter to only calculate prevalence over the past 60 days. When we make that adjustment, we see there are a lot more countries that haven’t sequenced any samples in the last two months (show in grey), and overall, the prevalence of P.1 has been dropping as the Delta lineages spread across the world.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1_by_country_60days</span> <span class="op">=</span> <span class="fu"><a href="../reference/getCumulativeBySubadmin.html">getCumulativeBySubadmin</a></span><span class="op">(</span>pangolin_lineage <span class="op">=</span> <span class="st">"P.1"</span>, ndays <span class="op">=</span> <span class="fl">60</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span>


<span class="co"># join by the ISO3 ID</span>
<span class="va">gadm_p1_60days</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">gadm</span>, <span class="va">p1_by_country_60days</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GID_0"</span> <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotChoropleth.html">plotChoropleth</a></span><span class="op">(</span><span class="va">gadm_p1_60days</span>, title<span class="op">=</span><span class="st">"Cumulative P.1 prevalence in the past 60 days"</span><span class="op">)</span></code></pre></div>
<p><img src="varianttracker_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="specifying-a-location">Specifying a location<a class="anchor" aria-label="anchor" href="#specifying-a-location"></a>
</h4>
<p>Instead of looking at the prevalence by country, you can also specify a location like Brazil, and look at the cumulative prevalence one administrative unit below that. So, for countries, you will get the prevalence at the state/province-level, and for states, you get the prevalnece at the county level. Note that the smaller the geographic area, it’s likely that the uncertainty in these estimates will increase, since there’s less sampling.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="co"># P.1 prevalence by Brazilian states</span>
<span class="va">p1_in_brazil</span> <span class="op">=</span> <span class="fu"><a href="../reference/getCumulativeBySubadmin.html">getCumulativeBySubadmin</a></span><span class="op">(</span>pangolin_lineage <span class="op">=</span> <span class="st">"P.1"</span>, location <span class="op">=</span> <span class="st">"Brazil"</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span>

<span class="co"># P.1 prevalence by Californian counties</span>
<span class="va">p1_in_california</span> <span class="op">=</span> <span class="fu"><a href="../reference/getCumulativeBySubadmin.html">getCumulativeBySubadmin</a></span><span class="op">(</span>pangolin_lineage <span class="op">=</span> <span class="st">"P.1"</span>, location <span class="op">=</span> <span class="st">"California"</span><span class="op">)</span>
<span class="co">#&gt; Retrieving data...</span></code></pre></div>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="what-do-we-know-about-the-variant-in-the-literature">What do we know about the variant in the literature?<a class="anchor" aria-label="anchor" href="#what-do-we-know-about-the-variant-in-the-literature"></a>
</h3>
<p>The pace of research on COVID-19 has been staggering, with the research community publishing thousands of papers, clinical trials, datasets and more each week. The <a href="https://outbreak.info/resources/search?q=%22P.1%22" class="external-link">P.1 papers and other resources</a> can be found in outbreak.info’s Research Library. At this time, the R package does not provide access to these data, but they can be found in our <a href="https://api.outbreak.info/try/resources" class="external-link">resource API</a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Manar Alkuzweny, Karthik Gangavarapu, Laura Hughes.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
